### PR TITLE
Feature Dependency and Impact Analysis: Impact Analysis & Data Retrieval API #84 (NEW)

### PR DESCRIPTION
**Pull request by DmitriyChernyakSetronica**
**Objective**
Develop the backend logic and API endpoints to retrieve dependency information and analyze the impact of feature changes. 🔍

**Requirements**
* **Dependency Retrieval:** Implement logic to fetch all dependencies (features this one relies on) and dependents (features that rely on this one).
* **Impact Analysis Logic:** Create a service that can identify all features potentially affected by a change to a given feature, including both direct and indirect dependents.
* **Filtering:** Support filtering the impact analysis results by parameters such as product, release, or status.

**API Endpoints**:
* `GET /features/{featureCode}/dependencies`: List all features that `{featureCode}` depends on.
* `GET /features/{featureCode}/dependents`: List all features that depend on `{featureCode}`.
* `GET /features/{featureCode}/impact`: Provide a comprehensive list of all affected features, with support for filtering via query parameters.

**Test Coverage**
* Tests for correct identification of all direct and indirect dependents.
* Tests for API endpoints, including the filtering logic.
* Tests for edge cases such as cyclic dependencies or functions without dependencies run without exceptions or hangs.

**Acceptance Criteria**
* The API correctly returns all dependencies and dependents for a given feature.
* The impact analysis endpoint accurately identifies the full scope of affected features.
* Filtering works as specified.

### CODE REVIEW

File: src/main/java/com/sivalabs/ft/features/domain/FeatureDependencyService.java [hunk 1]
```diff
@@ -70,4 +79,172 @@ public void deleteFeatureDependency(Commands.DeleteFeatureDependencyCommand cmd)
 
         featureDependencyRepository.delete(dependency);
     }
+
+    /**
+     * Get all dependencies for a given feature (features this one depends on)
+     */
+    @Transactional(readOnly = true)
+    public List<FeatureDto> getDependencies(String featureCode, String productCode) {
```
Comment (volkov) [inline src/main/java/com/sivalabs/ft/features/domain/FeatureDependencyService.java:87]: Unused?

File: src/main/java/com/sivalabs/ft/features/domain/FeatureDependencyService.java [hunk 2]
```diff
@@ -70,4 +79,172 @@ public void deleteFeatureDependency(Commands.DeleteFeatureDependencyCommand cmd)
...
+            String featureCode, String productCode, String releaseCode, FeatureStatus status) {
+        // Feature existence is checked in controller to avoid transaction rollback issues
+
+        List<FeatureDependency> dependencies =
+                featureDependencyRepository.findDependenciesWithFiltering(featureCode, productCode);
+
+        // Apply additional filtering in memory for release and status
+        if (releaseCode != null || status != null) {
+            dependencies = dependencies.stream()
+                    .filter(dep -> releaseCode == null
```
Comment (volkov) [inline src/main/java/com/sivalabs/ft/features/domain/FeatureDependencyService.java:105]: Should we change it to query in repo?

File: src/main/java/com/sivalabs/ft/features/domain/FeatureDependencyService.java (hunk 1 above)
Comment (DmitriyChernyakSetronica) [reply src/main/java/com/sivalabs/ft/features/domain/FeatureDependencyService.java]: fixed
